### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by JessSaavedra**
## Summary

- Added tests of models and controllers using Rspec

## Trello Card

https://trello.com/c/SyczsgiE

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -33,6 +33,10 @@ gem "less-rails" #Sprockets (what Rails 3.1 uses for its asset pipeline) support
 
 gem "twitter-bootstrap-rails"
 
+gem 'factory_girl_rails'
```
Comment (epintos) [inline Gemfile]: Always try to add a comment on top of the gem indicating why you include it

File: spec/controllers/articles_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  describe "GET #new" do
+    it "assigns new article to @article" do
+      get :new
+      expect(assigns(:article)).to be_a_new(Article)
+    end
+  end
+  describe "PUT #update" do
+  	context "with valid attributes" do
+      it "changes attributes" do
+        put :update, id: article, article: FactoryGirl.attributes_for(:article, title: "A title", text: "This is an amazing article. Please read it.")
```
Comment (epintos) [inline spec/controllers/articles_controller_spec.rb]: Its better to have the variables in a let to make the tests clearer

File: spec/controllers/articles_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        article.reload
+        expect(article.title).to eq("A title")
```
Comment (epintos) [inline spec/controllers/articles_controller_spec.rb]: also here you could to expect(article.title).to eq (article.title)

File: spec/controllers/comments_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        expect(response).to redirect_to article
+      end
+    end
+    context "with invalid attributes" do
+      it "does not save the comment" do
+        expect{ post :create, comment: FactoryGirl.attributes_for(:invalid_comment, :article_id => article) }.not_to change(Comment,:count)
+      end
+    end
+  end
+end
```
Comment (epintos) [inline spec/controllers/comments_controller_spec.rb]: Please setup yoour sublime to always add an enter at the end of file. I will send that setup later

**Comment by epintos** [comment]
Nice! Only a few comments

File: spec/controllers/articles_controller_spec.rb (hunk 3 above)
Comment (JessSaavedra) [inline spec/controllers/articles_controller_spec.rb]: I don't understand, wouldn't that always be true even if the article hadn't changed?

File: spec/controllers/articles_controller_spec.rb [hunk 5]
```diff
@@ -36,13 +37,13 @@
...
+        put :update, id: article, article: FactoryGirl.attributes_for(:another_article)
         article.reload
-        expect(article.title).to eq("A title")
-        expect(article.text).to eq("This is an amazing article. Please read it.")
+        expect(article.title).to eq(another_article.title)
+        expect(article.text).to eq(another_article.text)
       end
       it "redirects to updated article" do
-        put :update, id: article, article: FactoryGirl.attributes_for(:article, title: "A title", text: "This is an amazing article. Please read it.")
+        put :update, id: article, article: FactoryGirl.attributes_for(:another_article)
```
Comment (epintos) [inline spec/controllers/articles_controller_spec.rb]: This attirbutes could be in a let two, to avoid duplicating the invocation

**Comment by epintos** [comment]
:+1:

**Comment by JessSaavedra** [comment]
Merged manually

**Pull request closed without merging**